content/docs/code-splitting.md

```diff
@@ -151,6 +151,20 @@ This will automatically load the bundle containing the `OtherComponent` when thi
 
 `React.lazy` takes a function that must call a dynamic `import()`. This must return a `Promise` which resolves to a module with a `default` export containing a React component.
 
+> Note:
+>
+> You might be tempted to use `React.lazy` to declare your lazy loaded resource inside the component which works, but keep in mind that the resulting component would be remounted on every render of containing component.
+
+```js
+function MyComponent() {
+  return (
+    {/* Don't do this */}
+    <Route path="/foo" component={React.lazy(() => import('./OtherComponent'))} />
+  );
+}
+```
+
```

Comment:
Might it be worth explaining why the resulting component gets remounted - ie that the object handle to the lazy Component is changing / being recreated every render?

And can we have also give a solution? Perhaps in the form of a hook?

We should also avoid Memo, because it's not guaranteed not to change (unlike Ref which is). In this case,

```
/* Do this instead. */
function useLazy(callToImport) {
    // We store the handle to the created lazy component so we can use the same component on every render.
    const componentRef = useRef();
    if (!componentRef .current) {
        componentRef .current = React.lazy(callToImport);
    }
    return [componentRef .current];
}

function MyComponent() {
    const [OtherComponent] = useLazy(() => import('./OtherComponent'));
    return (
       <Route path="/foo" component={OtherComponent} />
    );
}
```